source/dub/recipe/packagerecipe.d

```diff
@@ -250,9 +250,17 @@ struct BuildSettingsTemplate {
 						continue;
 					}
 
-					foreach (d; dirEntries(path.toNativeString(), pattern, SpanMode.depth)) {
-						import std.path : baseName;
-						if (baseName(d.name)[0] == '.' || d.isDir) continue;
+					auto pstr = path.toNativeString();
+					foreach (d; dirEntries(pstr, pattern, SpanMode.depth)) {
+						import std.path : baseName, pathSplitter;
+						import std.algorithm.searching: canFind;
+						import std.string: startsWith;
+						// eliminate any hidden files, or files in hidden directories. But always include
+						// files that are listed inside hidden directories that are specifically added to
+						// the project.
+						if (baseName(d.name)[0] == '.' || d.isDir ||
```

Comment:
I have a feeling I can just remove the first part of this expression. But let's see if it passes first.

Reply:
sounds like a good idea to do now

Reply:
OK, I reworked it. I also realized that checking some weird directory like "a/b//c" could cause an array bounds error, so I added a check for that.